frontend/src/service/defenceService.ts

```diff
@@ -63,60 +65,61 @@ async function configureDefence(
 	return response.status === 200;
 }
 
-async function resetDefenceConfig(
-	defenceId: string,
-	configId: string
+async function resetDefenceConfigItem(
+	defenceId: DEFENCE_ID,
+	configItemId: DEFENCE_CONFIG_ITEM_ID,
+	level: LEVEL_NAMES
 ): Promise<DefenceResetResponse> {
 	const response = await sendRequest(`${PATH}resetConfig`, {
 		method: 'POST',
 		headers: {
 			'Content-Type': 'application/json',
 		},
-		body: JSON.stringify({ defenceId, configId }),
+		body: JSON.stringify({ defenceId, configItemId, level }),
 	});
 	return (await response.json()) as DefenceResetResponse;
 }
 
-function validatePositiveNumberConfig(config: string) {
+function validatePositiveNumberConfig(value: string) {
 	// config is a number greater than zero
-	return !isNaN(Number(config)) && Number(config) > 0;
+	return !isNaN(Number(value)) && Number(value) > 0;
 }
 
-function validateNonEmptyStringConfig(config: string) {
+function validateNonEmptyStringConfig(value: string) {
 	// config is non empty string and is not a number
-	return config !== '' && !Number(config);
+	return value !== '' && !Number(value);
 }
 
-function validateFilterConfig(config: string) {
+function validateFilterConfig(value: string) {
 	// config is not a list of empty commas
 	const commaPattern = /^,*,*$/;
-	return config === '' || !commaPattern.test(config);
+	return value === '' || !commaPattern.test(value);
```

Comment:
Ha, another unrelated thing!

This "comma pattern" will actually match the empty string. Pretty sure that's not what we wanted, seeing as we needed to specifically allow empty string to begin with.

We probably want `/^,+$/`, or better, we split the string on commas, trim() each entry, remove falsy values, and then check if the resulting list is empty or not.

Another to add to my todo list 😉

Reply:
that's easy enough to do here